[ws] Error thrown when close socket

### Description
<!-- Shortly describe your issue. -->
When close event is triggered in socket, error message will be thrown.

### Test Code
<!-- List the impacted test cases or sources code here. For example: -->
[WebSocketServer.js](https://github.com/01org/zephyr.js/blob/master/samples/websockets/WebSocketServer.js);
[WebSocketServer4.js](https://github.com/01org/zephyr.js/blob/master/samples/websockets/WebSocketServer4.js)
 
### Steps to Reproduction
<!-- Describe the exact steps how to reproduce the issue(if any precondition for this issue, please make sure put them firstly). -->
1. `make BOARD=frdm_k64f JS=samples/websockets/WebSocketServer.js`
2. `cp outdir/frdm_k64f/zephyr.bin /media/<username>/MBED/`
3. add route by using following command:
    ` ip -6 route add 2001:db8::/64 dev eno1`
4. execute `node samples/websockets/NodeWebSocketClient.js` in terminal
5. CTRL+C to stop connection

### Actual Result
<!-- Describe the actual behavior you get. -->
Got `MPU FAULT` error on K64F:
![selection_115](https://user-images.githubusercontent.com/19813546/28300044-5f31a1d8-6baf-11e7-95dc-adb2a1c52018.png)

Got `[ERROR] Cannot sent data to peer(-9)` error on QEMU:
![selection_114](https://user-images.githubusercontent.com/19813546/28300078-90438fac-6baf-11e7-9179-55473f0aeb38.png)

 ### Expected Result
<!-- Describe what the behavior would be without the issue. -->
No error thrown. 

### Test Builds
<!-- Include GitHub branch, commit id, target device, test date and result. If the result is fail, please use 'Fail', opposite is 'Pass',
     and use 'Crash' for crash issue. More than one builds can be listed here for history reference. -->
Branch | Commit Id | Target Device | Test Date | Result
---|---|---|---|---
master | af7eadd | FRDM-K64F | Jul 18, 2017 | Fail
master | af7eadd | QEMU | Jul 18, 2017 | Fail
 
### Additional Information
<!-- Specific test environments -->
<!-- Error logs, screenshots for the failure phenomenon and pass screenshot. -->
Websocket on A101 is failed by #1348 .

--- a/samples/websockets/WebSocketServer.js
+++ b/samples/websockets/WebSocketServer.js
@@ -24,12 +24,13 @@ var wss = new WebSocket.Server({
 });
 
 var count = 0;
+var t = null;
 
 wss.on('connection', function(ws) {
     console.log("CONNECTION");
     ws.on('message', function(message) {
         console.log("MESSAGE: " + message.toString('ascii'));
-        setTimeout(function() {
+        t = setTimeout(function() {
             ws.send(new Buffer("A MESSAGE"), true);
             if (count % 10 == 0) {
                 ws.ping(new Buffer("PING"));
@@ -46,8 +47,10 @@ wss.on('connection', function(ws) {
     });
     ws.on('close', function(code, reason) {
         console.log("close event: " + reason);
+        clearTimeout(t);
     });
     ws.on('error', function(error) {
         console.log(error.name + " : " + error.message);
+        clearTimeout(t);
     });
 });

--- a/samples/websockets/WebSocketServer4.js
+++ b/samples/websockets/WebSocketServer4.js
@@ -27,12 +27,13 @@ net_cfg.dhcp(function(address, subnet, gateway) {
     });
 
     var count = 0;
+    var t = null;
 
     wss.on('connection', function(ws) {
         console.log("CONNECTION");
         ws.on('message', function(message) {
             console.log("MESSAGE: " + message.toString('ascii'));
-            setTimeout(function() {
+            t = setTimeout(function() {
                 ws.send(new Buffer("A MESSAGE"), true);
                 if (count % 10 == 0) {
                     ws.ping(new Buffer("PING"));
@@ -49,9 +50,11 @@ net_cfg.dhcp(function(address, subnet, gateway) {
         });
         ws.on('close', function(code, reason) {
             console.log("close event: " + reason);
+            clearTimeout(t);
         });
         ws.on('error', function(error) {
             console.log(error.name + " : " + error.message);
+            clearTimeout(t);
         });
     });
 });

